### PR TITLE
Fix libc errno returns

### PR DESCRIPTION
**Pull request by Yaxuan-w**
## Description

Fixes [#317](https://github.com/Lind-Project/lind_project/issues/317)
 

### Type of change

Fixed problems with return values especially when syscall failed (errno related).
- shmdt
- munmap
- mprotect

- [x] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Corresponding test are in branch `test-sigprocmask` in `Lind-Project/tests/testcases`, and in format: testX.c for test of X.

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been added to a pull request and/or merged in other modules (native-client, lind-project, safeposix-rust)

### CODE REVIEW

**Comment by rennergade** [comment]
Did you run the test suites on this?

**Comment by yizhuoliang** [comment]
Make sense, neg signs are never need when return a call to another syscall, they are needed only when you set the errorno

**Comment by Yaxuan-w** [comment]
> Did you run the test suites on this?

Yes, it passes all test cases in /lind_project

File: configure.new
Comment (RusherRG) [inline configure.new:1]: I think you committed it by mistake. Can you remove this file?

**Review by RusherRG — APPROVED** [review]
The changes look self-explanatory. We are returning -1 for errors and 0 otherwise and setting the errno accordingly. :v:

File: configure.new
Comment (RusherRG) [inline configure.new:1]: I think you pushed this by mistake. Can you remove this file?

**Review by RusherRG — CHANGES_REQUESTED** [review]

File: configure.new
Comment (rennergade) [reply configure.new:1]: @Yaxuan-w alice it looks like you added this by mistake, can you remove? then i think this can be merged

Comment (Yaxuan-w) [reply configure.new:1]: Thanks for pointing out!

**Pull request merged**